File: python/solutions/0001_two_sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        seen : dict[int, int] = {}
        for i , num in enumerate(nums):
            need = target - num
            if need in seen:
                return [seen[need], i]
            seen[num] = i

        return []

File: cpp/solutions/0001_two_sum.cpp
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen;
        seen.reserve(nums.size());

        for (int i = 0; i < static_cast<int>(nums.size()); ++i) {
            const int need = target - nums[i];
            const auto it = seen.find(need);
            if (it != seen.end()) {
                return {it->second, i};
            }
            seen.emplace(nums[i], i);
        }

        return {};
    }
};
